db/crud.go

```diff
@@ -842,6 +842,204 @@ func (db *Database) PutExistingRev(docid string, body Body, docHistory []string,
 	return doc, err
 }
 
+func (db *Database) validateExistingDoc(doc *Document, importAllowed, docExists bool) error {
+	if !importAllowed && docExists && !doc.HasValidSyncData(db.writeSequences()) {
+		return base.HTTPErrorf(409, "Not imported")
+	}
+	return nil
+}
+
+func (db *Database) validateNewBody(body Body) error {
```

Comment:
Doesn't need to be a method on Database - nothing uses it

```suggestion
func validateNewBody(body Body) error {
```